kong/templates/nginx_kong.lua

```diff
@@ -63,6 +63,11 @@ upstream kong_upstream {
     keepalive ${{UPSTREAM_KEEPALIVE}};
 }
 
+map $http_upgrade $connection_upgrade {
+    default upgrade;
+    '' close;
```

Comment:
So if I understand correctly, this will set the headers to be sent to upstream as:
- `Connection: upgrade` by default
- `Connection: close` if the downstream `Upgrade` header is not set.

If so, this is a no-no. Switching to `HTTP/1.1` is a good change (which we should have done a while ago but somehow never did. Properly managing the `Connection` header is also something that we somehow never implemented (very bad, proxies and gateways should *never* forward it to upstream services). But this is too rushed and not thought through.

In `HTTP/1.1`, connections are considered persistent by default. This means that unless a `Connection: close` header is sent from a client, the server will consider the connection as being persistent. Now, the problem here is that the defined behavior is to send `Connection: close` as soon as no `Upgrade` header is received from downstream, which means: no persistent connections, ever.

I believe a better solution could be:
```nginx
map $http_upgrade $connection_upgrade {
    close       close;
    upgrade  upgrade;
    default     keep-alive;
}
```

To be tested of course.